Core/include/Acts/AmbiguityResolution/AthenaAmbiguityResolution.ipp

```diff
@@ -0,0 +1,406 @@
+                             << " has score = 0, due to optional cuts.");
+        break;
+      }
+    }
+
+    if (score == 0) {
+      iTrack++;
+      trackScore.push_back(score);
+      ACTS_DEBUG("Track: " << iTrack << " score : " << score);
+      continue;
+    }
+    // Reject tracks which didn't pass the detector cuts.
+    for (std::size_t detectorId = 0; detectorId < m_cfg.detectorMap.size();
+         detectorId++) {
+      auto detector_it = m_cfg.detectorMap.find(detectorId);
+      auto detector = detector_it->second;
+
+      ACTS_DEBUG("---> Found summary information");
+      ACTS_DEBUG("---> Detector ID: " << detectorId);
+      ACTS_DEBUG("---> Number of hits: " << trackFeaturesMap[detectorId].nHits);
+      ACTS_DEBUG(
+          "---> Number of holes: " << trackFeaturesMap[detectorId].nHoles);
+      ACTS_DEBUG("---> Number of outliers: "
+                 << trackFeaturesMap[detectorId].nOutliers);
+
+      if ((trackFeaturesMap[detectorId].nHits < detector.minHits) ||
+          (trackFeaturesMap[detectorId].nHits > detector.maxHits) ||
+          (trackFeaturesMap[detectorId].nHoles > detector.maxHoles) ||
+          (trackFeaturesMap[detectorId].nOutliers > detector.maxOutliers)) {
+        score = 0;
+        ACTS_DEBUG("Track: " << iTrack
+                             << " has score = 0, due to detector cuts");
+        break;
+      }
+    }
+
+    if (score == 0) {
+      iTrack++;
+      trackScore.push_back(score);
+      ACTS_DEBUG("Track: " << iTrack << " score : " << score);
+      continue;
+    }
+
+    // real scoring starts here
+
+    if (!m_cfg.useAmbiguityFunction) {
+      // if the ambiguity function is used, the score is processed with a
+      // different algorithm than the simple score.
+      score = 100;
+      // Adding the score for each detector.
+      // detector score is determined by the number of hits/hole/outliers *
+      // hit/hole/outlier scoreWeights in a detector.
+      for (std::size_t detectorId = 0; detectorId < m_cfg.detectorMap.size();
+           detectorId++) {
+        auto detector_it = m_cfg.detectorMap.find(detectorId);
+        auto detector = detector_it->second;
+        score += trackFeaturesMap[detectorId].nHits * detector.hitsScoreWeight;
+        score +=
+            trackFeaturesMap[detectorId].nHoles * detector.holesScoreWeight;
+        score += trackFeaturesMap[detectorId].nOutliers *
+                 detector.outliersScoreWeight;
+        score += trackFeaturesMap[detectorId].nSharedHits *
+                 detector.otherScoreWeight;
+      }
+
+      // Adding scores based on optional weights
+      for (const auto& ambiweights : optionalCuts.weights) {
+        ambiweights(track, score);
+      }
+
+      // Adding the score based on the chi2/ndf
+      if (track.chi2() > 0 && track.nDoF() > 0) {
+        double p = 1. / log10(10. + 10. * track.chi2() / track.nDoF());
+        if (p > 0) {
+          score += p;
+        } else
+          score -= 50;
```

Comment:
```suggestion
        } else {
          score -= 50;
        }
```

also: magic number?

Reply:
I took this numbers from Athena ambiguity Resolution.  But the logic is by default all valid tracks have a starting score of 100 and based on the number of hits,holes etc we will either add or subtract from this number. so 50 is in the same order as 100.  Hence the choice.